src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt

```diff
@@ -208,12 +210,46 @@ class AssessmentService(
     return updateEpisode(episode, updatedEpisodeAnswers)
   }
 
-  @Transactional
   fun addEpisodeTableRow(
     assessmentUuid: UUID,
     episodeUuid: UUID,
     tableName: String,
     newTableRow: UpdateAssessmentEpisodeDto
+  ): AssessmentEpisodeDto {
+    return modifyEpisodeTableRow(
+      assessmentUuid,
+      episodeUuid,
+      tableName
+    ) { existingTable ->
+        extendTableAnswers(existingTable, newTableRow.answers)
+    }
+  }
+
+  fun updateEpisodeTableRow(
+    assessmentUuid: UUID,
+    episodeUuid: UUID,
+    tableName: String,
+    index: Int,
+    updatedTableRow: UpdateAssessmentEpisodeDto
+  ): AssessmentEpisodeDto {
+    return modifyEpisodeTableRow(
+      assessmentUuid,
+      episodeUuid,
+      tableName
+    ) { existingTable ->
+      if ((index < 0) || (index >= existingTable.values.first().size))
+        throw IllegalStateException("Bad index $index for table $tableName")
+
+      updateTableAnswers(existingTable, index, updatedTableRow.answers)
+    }
+  }
+
+  @Transactional
+  private fun modifyEpisodeTableRow(
```

Comment:
It seems that Spring needs to be able to override this method and that won't be possible if is private ? - `Methods annotated with '@Transactional' must be overridable`
![image](https://user-images.githubusercontent.com/9938931/117676571-92782680-b1a5-11eb-9ea3-8eba75a1ac5f.png)